crates/partition-store/src/partition_store.rs

```diff
@@ -415,6 +423,11 @@ impl PartitionStore {
                 )
             });
 
+        let snapshot = match read_isolation {
+            ReadIsolation::None => None,
+            ReadIsolation::Snapshot => Some(self.rocksdb.inner().as_raw_db().snapshot()),
```

Comment:
I didn't run any benchmarks on the negative effects of using snapshot reads in the invoker yet.